src/ucs/sys/sock.c

```diff
@@ -132,8 +132,11 @@ static ucs_status_t ucs_socket_check_errno(int io_errno)
         return UCS_ERR_NO_PROGRESS;
     }
 
-    if ((io_errno == ECONNRESET) || (io_errno == ECONNREFUSED)) {
-        return UCS_ERR_NOT_CONNECTED; /* Connection reset by peer */
+    if (/* Connection reset by peer */
+        (io_errno == ECONNRESET) || (io_errno == ECONNREFUSED) ||
+        /* Connection establishment procedure timed out */
+        (io_errno == ETIMEDOUT)) {
```

Comment:
why did you add this? timeout seems like a different error - PEER_FAILURE?
we can check io_status==PEER_FAILURE in uct_tcp_ep_handle_dropped_connect

Reply:
> we can check io_status==PEER_FAILURE in uct_tcp_ep_handle_dropped_connect

done, return `UCS_ERR_TIMED_OUT` in case of `ETIMEDOUT` error